Source/Particles/Collision/BinaryCollision/NuclearFusion/ProtonBoronFusionInitializeMomentum.H

```diff
@@ -118,9 +118,9 @@ namespace {
         const amrex::ParticleReal p2y = u2y * m2;
         const amrex::ParticleReal p2z = u2z * m2;
         // Square norm of the total (sum between the two particles) momenta in the lab frame
```

Comment:
While we are here, let's optimize these device operations. Multiplications are very cheap compared to calling `std::pow`:
```suggestion
        // Square norm of the total (sum between the two particles) momenta in the lab frame
        auto constexpr quad = [](double const x) { return x*x; };
```

cc fyi: @NeilZaim 

cc @WeiqunZhang we could add `square`/`quad` and `cubic` to `AMReX_Math.H` later on, this is universally useful (that's what we did in PIConGPU/PMacc/Alpaka at some point, when we optimized kernels.) What do you think?
For the PR here, let's keep the inline definitions for now.

Reply:
I had the same thought. I would have used names like `pow2` and `pow3` to use a name consistent with `std::pow`.